clickhouse/client.h

```diff
@@ -273,6 +273,45 @@ class Client {
     void Insert(const std::string& table_name, const Block& block);
     void Insert(const std::string& table_name, const std::string& query_id, const Block& block);
 
+    /**
+     * @brief Begin and continuously send data for an \p INSERT query
+     * @param query The \p INSERT query must end with \p VALUES but contain no values
+     * @return The Block to use to send data to insert
+     *
+     * Use this API to insert multiple blocks of data into a table. Example:
+     *
+     * \code{.cpp}
+     * // Start the insertion.
+     * auto block = client->BeginInsert("INSERT INTO foo (id, name) VALUES");
+     * 
+     * // Grab the columns from the inserter's block.
+     * auto block = inserter->GetBlock();
+     * auto col1 = (*block)[0]->As<ColumnUInt64>();
+     * auto col2 = (*block)[1]->As<ColumnString>();
+     *
+     * // Add a couple of records to the block.
+     * col1.Append(1);
+     * col1.Append(2);
+     * col2.Append("holden");
+     * col2.Append("naomi");
+     *
+     * Send those records.
+     * client->InsertData(*block);
+     *
+     * // Add another record.
+     * col1.Append(3);
+     * col2.Append("amos");
+     *
+     * // Send it and finish.
+     * client->EndInsert(*block);
+     * \endcode
+     */
+    std::unique_ptr<Block> BeginInsert(const std::string& query);
+    std::unique_ptr<Block> BeginInsert(const std::string& query, const std::string& query_id);
+    void InsertData(Block& block);
```

Comment:
Holler if you'd rather pass a `std::unique_ptr<Block>`. Seems okay to me to pass a `*block` instead, but I'm not yet up to snuff on idiomatic C++.

Reply:
But `InsertData` does modify the block, by design:

```cpp
void Client::Impl::InsertData(Block& block) {
    assert(inserting);
    block.RefreshRowCount();
    SendData(block);
    block.Clear();
}
```

Would you rather that refreshing the count and clearing be done by the caller?

Reply:
Switched to returning a `Block` in 0a3da16, and also moved the docs to the README. [Diff](https://github.com/ClickHouse/clickhouse-cpp/compare/fb148753cea70e80d0af952ce7b91a5fd01954c6..0a3da16ed87ce0f647b16e43ed8af56f68064242).

Reply:
Does the caller need to `delete` the returned `Block`? Seems like yes, since IIUC `Block()` creates it on the heap, yes?